Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -1353,8 +1360,11 @@ private void TryWaitForReconnect()
 
             if (!isResetTime)
             {
-                if (_previouslyInResetTime)
+                if (_stateManager.PreviouslyInResetTime)
                 {
+                    // do not restart after IBAutomater errors
+                    CheckIbAutomaterErrors();
```

Comment:
Maybe we could move this directly into the `ResetGatewayConnection()`, so we avoid having it spread and we don't have to rely on the caller to do the right thing.

Reply:
Hmm, the reason for having this check in the caller was to avoid throwing exceptions from the reset handler thread,
also, we have four calls to `CheckIbAutomaterErrors` where only two of them are before triggering the reset event.

Reply:
Right 👍  although the reset handler thread is already calling `CheckIbAutomaterErrors()` after restarting the ibAutomater.

Reply:
Correct, this should be addressed in the new version of `CheckIbAutomaterErrors()`, which now also sets the `algorithm.RuntimeError` via the `BrokerageMessageEvent`. The version in the current master only throws the exception which is logged in the reset handler thread, but the algorithm keeps running.